src/main/scala/org/ergoplatform/nodeView/wallet/WalletScanLogic.scala

```diff
@@ -63,46 +66,91 @@ object WalletScanLogic extends ScorexLogging {
                             walletVars: WalletVars,
                             height: Int,
                             blockId: ModifierId,
-                            transactions: Seq[ErgoTransaction]): (WalletRegistry, OffChainRegistry) = {
-
-    //todo: inefficient for wallets with many outputs, replace with a Bloom/Cuckoo filter?
-    //todo: at least, cache in RAM instead of reading from DB
-    val previousBoxIds = registry.allUnspentBoxes().map(tb => encodedBoxId(tb.box.id))
+                            transactions: Seq[ErgoTransaction],
+                            cachedOutputsFilter: Option[BloomFilter[Array[Byte]]]
+                           ): (WalletRegistry, OffChainRegistry, BloomFilter[Array[Byte]]) = {
```

Comment:
Please describe in ScalaDoc all parameters and the result.

Reply:
done